---
title: "Multiplexed identification of TF-specific reporters"
author: 
  - name: "Max Trauernicht"
    email: "m.trauernicht@nki.nl"
    affiliation: "Netherlands Cancer Institute - van Steensel lab"

date: '`r format(Sys.time(), "%d/%m/%Y")`'
output:
  html_document:
    theme: united
    highlight: pygments
    fig_caption: yes
    code_folding: hide
    df_print: kable
    toc: true
    toc_depth: 4
    toc_float:
      collapsed: false
---

---

### Aim
~36,000 reporters for 86 TFs were transfected into 9 different cell types + treated with ~90 different siRNAs or various signaling molecules. In this script I will analyze TF reporter activities in detail and review how individual reporters respond to TF concentration variations.

---

## Setup {.tabset}

<!-- little HTML script to do indentation of the table of contents -->
<script>
    $(document).ready(function() {
      $items = $('div#TOC li');
      $items.each(function(idx) {
        num_ul = $(this).parentsUntil('#TOC').length;
        $(this).css({'text-indent': num_ul * 10, 'padding-left': 0});
      });

    });
</script>

```{css, echo = FALSE}
div.sourceCode {
  overflow-x: hidden;
}
```

### Libraries 

```{r setup, out.width= "80%", fig.align= "center", warning = FALSE, message= FALSE}
knitr::opts_chunk$set(echo = TRUE)
StartTime <-Sys.time()

# 8-digit Date tag:
Date <- substr(gsub("-","",Sys.time()),1,8) 
# libraries:
library(RColorBrewer)
library(ggplot2)
library(dplyr)
library(maditr)
library(tibble)
library(pheatmap)
library(ggpubr)
library(visNetwork)
library(ggbeeswarm)
library(ggforce)
library(viridis)
library(plyr)
library(igraph)
library(ggraph)
library(cowplot)
library(gridExtra)
library(pROC)
library(tidyr)
library(stringr)
library(randomForest)
library(ggrastr)
library(readr)
library(ggbiplot)
library(IHW)
library(biomaRt)
library(ggh4x)
library(ggiraph)
library(plotly)
library(AMR)
library(umap)
library(ggrepel)
library(chromVAR)
library(motifmatchr)
library(Matrix)
library(SummarizedExperiment)
library(BiocParallel)
library(JASPAR2016)
library(BSgenome.Mmusculus.UCSC.mm10)
library(BSgenome.Hsapiens.UCSC.hg38)
```

### Functions

```{r out.width= "80%", fig.align= "center"}
### Custom functions
SetFileName <- function(filename, initials) {
  # Set filename with extension and initials to make filename with date integrated.
  filename <- substitute(filename)
  initials <- substitute(initials)
  filename <- paste0(initials, Date, filename)
  filename
}


# Extract p-value from linear model
lmp <- function (modelobject) {
    if (class(modelobject) != "lm") stop("Not an object of class 'lm' ")
    f <- summary(modelobject)$fstatistic
    p <- pf(f[1],f[2],f[3],lower.tail=F)
    attributes(p) <- NULL
    return(p)
}

# Set custom ggplot2 theme and custom colors
theme_classic_lines <- function() {
  theme_pubr(border = F, legend = "top") +
            theme(panel.grid.major = element_line(colour = "#adb5bd", linewidth = 0.25),
                  strip.background = element_rect(fill = "#ced4da")
            )
}

theme_classic_lines_45 <- function() {
  theme_pubr(border = T, legend = "top", x.text.angle = 45) +
            theme(panel.grid.major = element_line(colour = "#adb5bd", linewidth = 0.25),
                  strip.background = element_rect(fill = "#ced4da")
            )
}

theme_classic_lines_90 <- function() {
  theme_pubr(border = T, legend = "top", x.text.angle = 90) +
            theme(panel.grid.major = element_line(colour = "#adb5bd", linewidth = 0.25),
                  strip.background = element_rect(fill = "#ced4da")
            )
}

theme_barplot <- function() {
  theme_pubr(border = T, legend = "none") +
            theme(panel.grid.major.x = element_line(colour = "black", linewidth = 0.25),
                  strip.background = element_blank(),
                  strip.text = element_text(face="bold", hjust=0)
            )
}

theme_set(theme_classic_lines())

colors_diverse <- c("#264653", "#9AC1AE", "#5D987B", "#f2cc8f", "#e76f51")

cell_colors <- c("A549" = "#f2cc8f", "HCT116" = "#ED1C24", "HEK293" = "#00B4D8", 
                               "HepG2" = "#B3B3B3", "K562" = "#A67C52", "MCF7" = "#81B29A", 
                               "U2OS" = "#3D405B", "mES" = "#EAB69F", "NPC" = "#E07A5F")

ggplot_custom <- function(...) ggplot2::ggplot(...) + 
  scale_color_manual(values = colors_diverse) + 
  scale_fill_manual(values = colors_diverse)
```

### Loading data

```{r out.width= "80%", fig.align= "center"}
# Import TF reporter activities (only of the identified prime reporters)
## Import reporter activities
cDNA_df <- read.csv("/DATA/usr/m.trauernicht/projects/ATAC_TF_reporter_comparison/TF_reporter_analyses/mt20240724_reporter_activity_filt_combined_complete.csv", header = T) %>%
  mutate(tf = gsub("_.*", "", tf))

# Load RNA-seq data
tf_rna <- read_tsv("/DATA/usr/m.trauernicht/data/RNA_seq/rna_tpm_all_tfs.tsv")

## Import prime reporters
prime_reporters <- read_csv("/DATA/usr/m.trauernicht/projects/SuRE-TF/data/best_reporters.csv") %>%
  filter(conf_level > 1)

prime_activities <- cDNA_df %>%
  #filter(reporter_id %in% prime_reporters$reporter_id) %>%
  distinct(tf, condition, reporter_activity_dif_sample, replicate) %>%
  mutate(reporter_activity_dif_sample = ave(reporter_activity_dif_sample, tf, condition, replicate, FUN = function(x) mean(x, na.rm = T))) %>%
  distinct() %>%
  mutate(replicate = toupper(replicate)) %>%
  filter(condition %in% c("mES_SOX2_DEG", "mES_POU5F1_DEG", "mES_FOXA1_OE", "mES_PD", "mES_CH", "mES_LIF", "U2OS_PMA", "U2OS_Heat", "U2OS_Calcitriol", "MCF7_Hexestrol"))


# Import peak count information for ChromVAR
peak_count <- readRDS("/DATA/usr/m.trauernicht/projects/ATAC_TF_reporter_comparison/ATAC_seq_analyses/rds/experiment_7784_20240724.rds")
assayNames(peak_count) <- "counts"

# Add GC bias
peak_count <- addGCBias(peak_count, genome = BSgenome.Mmusculus.UCSC.mm10)

## Filter peaks - each peak must have at least one count in one condition
peak_count <- filterPeaks(peak_count, non_overlapping = TRUE)

# Import motif information
seq_logos <- readRDS("/DATA/usr/m.trauernicht/projects/SuRE-TF/data/library_design/lambert_2018/seq_logos.rds")
motifs_selected <- readRDS("/DATA/usr/m.trauernicht/projects/SuRE-TF/data/library_design/lambert_2018/motifs_selected.rds") %>%
  mutate(tf = toupper(tf)) %>%
  mutate(tf = ifelse(tf == "AHR:ARNT", "AHR::ARNT", tf)) %>%
  mutate(tf = ifelse(tf == "RAR:RXR", "RARA:RXRA", tf)) %>%
  rbind(data.frame(tf = c("STAT1", "STAT4", "ATF2", "ATF4", "ATF6", "CLOCK"),
                   motif_id = c("M06794_1.94d.txt", "M06797_1.94d.txt", "M08481_1.94d.txt", "M08489_1.94d.txt", "M09435_1.94d.txt", "M05596_1.94.txt")))
  
seq_logos_pwm <- list()
for (i in names(seq_logos)) {
  if (i %in% motifs_selected$motif_id) {
  seq_logos_pwm[[i]] <- universalmotif::convert_motifs(seq_logos[[i]], class = "TFBSTools-PWMatrix")
  seq_logos_pwm[[i]]@name <- i
  seq_logos_pwm[[i]]@strand <- "+-"
  }
}
# 
# seq_logos_pwm <- list()
# for (i in names(seq_logos)) {
#   seq_logos_pwm[[i]] <- universalmotif::convert_motifs(seq_logos[[i]], class = "TFBSTools-PWMatrix")
#   name_tf <- all_motifs$tf[all_motifs$motif_id == i]
#   seq_logos_pwm[[i]]@name <- name_tf
#   seq_logos_pwm[[i]]@strand <- "+-"
# }
# 

seq_logos_pwm_all <- do.call(TFBSTools::PWMatrixList, seq_logos_pwm)

motif_ix <- matchMotifs(seq_logos_pwm_all, peak_count, 
                        genome = BSgenome.Mmusculus.UCSC.mm10)
```

---

### Run chromVAR
```{r out.width= "80%", fig.align= "center"}
# Compute deviations
#bg <- getBackgroundPeaks(object = sox2_peak_count_gc_filt)
dev <- computeDeviations(object = peak_count, annotations = motif_ix
                         #,background_peaks = bg
                         )

dev_dif <- differentialDeviations(dev, group = peak_count@colData$treatment)

dev_scores <- deviationScores(dev) %>%
  as.data.frame() %>%
  rownames_to_column(var = "motif_id") %>%
  left_join(motifs_selected) %>%
  pivot_longer(cols = -c(motif_id, tf), names_to = "sample", values_to = "chromVAR_zscore") %>%
  mutate(replicate = gsub(".*(R[1-3]{1}$)", "\\1", sample)) %>%
  mutate(condition = gsub("_R[1-3]{1}$", "", sample)) %>%
  mutate(chromVAR_zscore_mean = ave(chromVAR_zscore, motif_id, condition, FUN = mean)) %>%
  distinct(motif_id, tf, condition, replicate, chromVAR_zscore_mean, chromVAR_zscore)

dev_scores_ctrl <- dev_scores %>%
  filter(condition %in% c("mES_2i_LIF", "mES_FOXA1_ctrl", "mES_POU5F1_ctrl", "mES_SOX2_ctrl")) %>%
  dplyr::select("ref_condition" = condition, "ctrl_zscore" = chromVAR_zscore, tf, replicate) 

condition_match <- data.frame(condition = c("mES_LIF", "mES_PD", "mES_CH", "mES_FOXA1_OE", "mES_POU5F1_DEG", "mES_SOX2_DEG"),
                              ref_condition = c("mES_2i_LIF", "mES_2i_LIF", "mES_2i_LIF", "mES_FOXA1_ctrl", "mES_POU5F1_ctrl", "mES_SOX2_ctrl"))

dev_scores_dif <- dev_scores %>%
  left_join(condition_match) %>%
  left_join(dev_scores_ctrl) %>%
  mutate(chromVAR_zscore_dif = chromVAR_zscore - ctrl_zscore) %>%
  mutate(chromVAR_zscore_dif_mean = ave(chromVAR_zscore_dif, tf, condition, FUN = mean))

# Compute variability
variability <- computeVariability(dev)

plotVariability(variability, use_plotly = FALSE)
```

---

## Correlate inferred activity with measured activity
```{r}
## Correlate chromVAR activities with measured activity
tf_activity_chromVAR_dev <- prime_activities %>%
  left_join(tf_rna %>% filter(cell == "mES") %>% distinct(tf, nTPM)) %>%
  left_join(dev_scores_dif %>% distinct(tf, condition, chromVAR_zscore_dif, replicate) #%>% mutate(replicate = ifelse(replicate == "R2", "R3", "R2"))
            ) %>%
  distinct(tf, reporter_activity_dif_sample, chromVAR_zscore_dif, replicate, condition, nTPM) %>%
  na.omit()#%>%
  #filter(nTPM > .1 | tf %in% c("SOX9", "FOXA1"))

ggplot(tf_activity_chromVAR_dev,
       aes(x = chromVAR_zscore_dif, y = reporter_activity_dif_sample)) +
  geom_text_repel(aes(label = tf), box.padding = 0.5) +
  geom_point() +
  # add linear regression line and correlation coefficient
  geom_smooth(method = "lm", color = "red") +
  geom_hline(yintercept = 0, linetype = "dashed") +
  geom_vline(xintercept = 0, linetype = "dashed") +
  labs(x = "chromVAR deviation z-score dif", y = "Reporter activity (FC)") +
  theme_pubr(border = T) +
  facet_grid(condition~replicate)
```


---

## ChromVAR on the human conditions

### Loading data

```{r out.width= "80%", fig.align= "center"}
# Import peak count information for ChromVAR
peak_count <- readRDS("/DATA/usr/m.trauernicht/projects/ATAC_TF_reporter_comparison/ATAC_seq_analyses/rds/experiment_7784_20240724_human.rds")
assayNames(peak_count) <- "counts"

# Add GC bias
peak_count <- addGCBias(peak_count, genome = BSgenome.Hsapiens.UCSC.hg38)

## Filter peaks - each peak must have at least one count in one condition
peak_count <- filterPeaks(GenomicRanges::sort(peak_count), non_overlapping = TRUE)

motif_ix <- matchMotifs(seq_logos_pwm_all, peak_count, 
                        genome = BSgenome.Hsapiens.UCSC.hg38)
```

---

### Run chromVAR
```{r out.width= "80%", fig.align= "center"}
# Compute deviations
#bg <- getBackgroundPeaks(object = sox2_peak_count_gc_filt)
dev <- computeDeviations(object = peak_count, annotations = motif_ix
                         #,background_peaks = bg
                         )

dev_dif <- differentialDeviations(dev, group = peak_count@colData$treatment)

dev_scores <- deviationScores(dev) %>%
  as.data.frame() %>%
  rownames_to_column(var = "motif_id") %>%
  left_join(motifs_selected) %>%
  pivot_longer(cols = -c(motif_id, tf), names_to = "sample", values_to = "chromVAR_zscore") %>%
  mutate(replicate = gsub(".*(R[1-3]{1}$)", "\\1", sample)) %>%
  mutate(condition = gsub("_R[1-3]{1}$", "", sample)) %>%
  mutate(chromVAR_zscore_mean = ave(chromVAR_zscore, motif_id, condition, FUN = mean)) %>%
  distinct(motif_id, tf, condition, replicate, chromVAR_zscore_mean, chromVAR_zscore)

dev_scores_ctrl <- dev_scores %>%
  filter(condition %in% c("U2OS_DMSO", "MCF7_DMSO")) %>%
  dplyr::select("ref_condition" = condition, "ctrl_zscore" = chromVAR_zscore, tf, replicate) 

condition_match <- data.frame(condition = c("U2OS_PMA", "U2OS_Heat", "U2OS_Calcitriol", "MCF7_Hexestrol"),
                              ref_condition = c("U2OS_DMSO", "U2OS_DMSO", "U2OS_DMSO", "MCF7_DMSO"))

dev_scores_dif <- dev_scores %>%
  left_join(condition_match) %>%
  left_join(dev_scores_ctrl) %>%
  mutate(chromVAR_zscore_dif = chromVAR_zscore - ctrl_zscore) %>%
  mutate(chromVAR_zscore_dif_mean = ave(chromVAR_zscore_dif, tf, condition, FUN = mean))

# Compute variability
variability <- computeVariability(dev)

plotVariability(variability, use_plotly = FALSE)
```

---

## Correlate inferred activity with measured activity
```{r}
## Correlate chromVAR activities with measured activity
tf_activity_chromVAR_dev <- prime_activities %>%
  #left_join(tf_rna %>% filter(cell %in% c("U2OS", "MCF7")) %>% distinct(tf, nTPM)) %>%
  left_join(dev_scores_dif %>% distinct(tf, condition, chromVAR_zscore_dif, replicate) #%>% mutate(replicate = ifelse(replicate == "R2", "R3", "R2"))
            ) %>%
  distinct(tf, reporter_activity_dif_sample, chromVAR_zscore_dif, replicate, condition) %>%
  na.omit()#%>%
  #filter(nTPM > .1 | tf %in% c("SOX9", "FOXA1"))

ggplot(tf_activity_chromVAR_dev,
       aes(x = chromVAR_zscore_dif, y = reporter_activity_dif_sample)) +
  geom_text_repel(aes(label = tf), box.padding = 0.5) +
  geom_point() +
  # add linear regression line and correlation coefficient
  geom_smooth(method = "lm", color = "red") +
  geom_hline(yintercept = 0, linetype = "dashed") +
  geom_vline(xintercept = 0, linetype = "dashed") +
  labs(x = "chromVAR deviation z-score dif", y = "Reporter activity (FC)") +
  theme_pubr(border = T) +
  facet_grid(condition~replicate)

ggplot(tf_activity_chromVAR_dev %>%
         mutate(chromVAR_zscore_dif_mean = ave(chromVAR_zscore_dif, tf, condition, FUN = mean)) %>%
         mutate(reporter_activity_dif_sample_mean = ave(reporter_activity_dif_sample, tf, condition, FUN = mean)) %>%
         distinct(tf, condition, chromVAR_zscore_dif_mean, reporter_activity_dif_sample_mean),
       aes(x = chromVAR_zscore_dif_mean, y = reporter_activity_dif_sample_mean)) +
  geom_text_repel(aes(label = tf), box.padding = 0.5) +
  geom_point() +
  # add linear regression line and correlation coefficient
  geom_smooth(method = "lm", color = "red") +
  geom_hline(yintercept = 0, linetype = "dashed") +
  geom_vline(xintercept = 0, linetype = "dashed") +
  labs(x = "chromVAR deviation z-score dif", y = "Reporter activity (FC)") +
  theme_pubr(border = T) +
  facet_wrap(~condition)
```
















---

## diffTF: Calculate TF activities based on ATAC+RNA-seq
```{r}

```










<!-- ## HEPG2 data from Tea -->

<!-- ```{r out.width= "80%", fig.align= "center"} -->
<!-- # Import peak count information for ChromVAR -->
<!-- peak_count <- readRDS("/DATA/usr/t.filipovska/projects/ATAC-seq_TF-footprinting/analysis/rds/experiment_6420_20210610.rds") -->
<!-- assayNames(peak_count) <- "counts" -->

<!-- ## Add GC bias -->
<!-- peak_count <- addGCBias(peak_count, genome = BSgenome.Hsapiens.UCSC.hg38) -->

<!-- ## Filter peaks - each peak must have at least one count in one condition -->
<!-- peak_count <- filterPeaks(peak_count, non_overlapping = TRUE) -->

<!-- motif_ix <- matchMotifs(seq_logos_pwm_all, peak_count,  -->
<!--                         genome = BSgenome.Hsapiens.UCSC.hg38) -->
<!-- ``` -->

<!-- --- -->

<!-- ### Run chromVAR -->
<!-- ```{r out.width= "80%", fig.align= "center"} -->
<!-- # Compute deviations -->
<!-- #bg <- getBackgroundPeaks(object = sox2_peak_count_gc_filt) -->
<!-- dev <- computeDeviations(object = peak_count, annotations = motif_ix -->
<!--                          #,background_peaks = bg -->
<!--                          ) -->

<!-- dev_dif <- differentialDeviations(dev, group = peak_count@colData$treatment) -->

<!-- dev_scores <- deviationScores(dev) %>% -->
<!--   as.data.frame() %>% -->
<!--   rownames_to_column(var = "motif_id") %>% -->
<!--   left_join(motifs_selected) %>% -->
<!--   pivot_longer(cols = -c(motif_id, tf), names_to = "sample", values_to = "chromVAR_zscore") %>% -->
<!--   mutate(replicate = gsub(".*(R[1-3]{1}$)", "\\1", sample)) %>% -->
<!--   mutate(condition = gsub("_R[1-3]{1}$", "", sample)) %>% -->
<!--   mutate(chromVAR_zscore_mean = ave(chromVAR_zscore, motif_id, condition, FUN = mean)) %>% -->
<!--   distinct(motif_id, tf, condition, replicate, chromVAR_zscore_mean, chromVAR_zscore) %>% -->
<!--   filter(condition %in% c("K562_DMSO", "HepG2_DMSO", "HepG2_CDCA")) -->

<!-- dev_scores_ctrl <- dev_scores %>% -->
<!--   filter(condition %in% c("HepG2_DMSO")) %>% -->
<!--   dplyr::select("ref_condition" = condition, "ctrl_zscore" = chromVAR_zscore, tf, replicate)  -->

<!-- condition_match <- data.frame(condition = c("HepG2_CDCA"), -->
<!--                               ref_condition = c("HepG2_DMSO")) -->

<!-- dev_scores_dif <- dev_scores %>% -->
<!--   left_join(condition_match) %>% -->
<!--   left_join(dev_scores_ctrl) %>% -->
<!--   mutate(chromVAR_zscore_dif = chromVAR_zscore - ctrl_zscore) %>% -->
<!--   mutate(chromVAR_zscore_dif_mean = ave(chromVAR_zscore_dif, tf, condition, FUN = mean)) -->

<!-- # Compute variability -->
<!-- variability <- computeVariability(dev) -->

<!-- plotVariability(variability, use_plotly = FALSE) -->
<!-- ``` -->

<!-- ## Correlate inferred activity with measured activity -->
<!-- ```{r} -->
<!-- ## Correlate chromVAR activities with measured activity -->
<!-- prime_activities <- read.csv("/DATA/usr/m.trauernicht/projects/SuRE-TF/data/gcf7124_stimulations/results/mt20240528_reporter_activity_filt_combined.csv", header = T) %>% -->
<!--   mutate(tf = gsub("_.*", "", tf)) %>% -->
<!--   filter(reporter_id %in% prime_reporters$reporter_id) %>% -->
<!--   distinct(tf, condition, reporter_dif_minP) %>% -->
<!--   filter(condition %in% c("HepG2", "K562", "HepG2_CDCA")) %>% -->
<!--   na.omit() -->

<!-- dev_scores_dif$condition <- gsub("HepG2_CDCA", "HepG2_CDCA", dev_scores_dif$condition) -->
<!-- dev_scores_dif$condition <- gsub("HepG2_DMSO", "HepG2", dev_scores_dif$condition) -->
<!-- dev_scores_dif$condition <- gsub("K562_DMSO", "K562", dev_scores_dif$condition) -->

<!-- tf_activity_chromVAR_dev <- prime_activities %>% -->
<!--   left_join(dev_scores_dif %>% distinct(tf, condition, chromVAR_zscore_dif_mean)) %>% -->
<!--   distinct(tf, reporter_dif_minP, chromVAR_zscore_dif_mean, condition) -->

<!-- ggplot(tf_activity_chromVAR_dev, -->
<!--        aes(x = chromVAR_zscore_dif_mean, y = reporter_dif_minP)) + -->
<!--   geom_text_repel(aes(label = tf), box.padding = 0.5) + -->
<!--   geom_point() + -->
<!--   # add linear regression line and correlation coefficient -->
<!--   geom_smooth(method = "lm", color = "red") + -->
<!--   geom_hline(yintercept = 0, linetype = "dashed") + -->
<!--   geom_vline(xintercept = 0, linetype = "dashed") + -->
<!--   labs(x = "chromVAR deviation z-score dif", y = "Reporter activity (FC)") + -->
<!--   theme_pubr(border = T) -->

<!-- prime_activities_2 <- read.csv("/DATA/usr/m.trauernicht/projects/SuRE-TF/data/gcf7124_stimulations/results/mt20240528_reporter_activity_filt_combined.csv", header = T) %>% -->
<!--   mutate(tf = gsub("_.*", "", tf)) %>% -->
<!--   filter(reporter_id %in% prime_reporters$reporter_id) %>% -->
<!--   distinct(tf, condition, reporter_activity_minP) %>% -->
<!--   filter(condition %in% c("HepG2", "K562")) %>% -->
<!--   na.omit() #%>% -->
<!--   #mutate(reporter_activity_minP = log2(reporter_activity_minP)) -->

<!-- tf_activity_chromVAR_dev <- prime_activities_2 %>% -->
<!--   left_join(dev_scores_dif %>% distinct(tf, condition, chromVAR_zscore_mean)) %>% -->
<!--   distinct(tf, reporter_activity_minP, chromVAR_zscore_mean, condition) -->

<!-- ggplot(tf_activity_chromVAR_dev, -->
<!--        aes(x = chromVAR_zscore_mean, y = log2(reporter_activity_minP))) + -->
<!--   geom_text_repel(aes(label = tf), box.padding = 0.5) + -->
<!--   geom_point() + -->
<!--   # add linear regression line and correlation coefficient -->
<!--   geom_smooth(method = "lm", color = "red") + -->
<!--   geom_hline(yintercept = 0, linetype = "dashed") + -->
<!--   geom_vline(xintercept = 0, linetype = "dashed") + -->
<!--   labs(x = "chromVAR deviation z-score", y = "Reporter activity (FC)") + -->
<!--   theme_pubr(border = T) + -->
<!--   facet_wrap(~condition) -->

<!-- tf_activity_chromVAR_dev_norm <- tf_activity_chromVAR_dev %>% -->
<!--   pivot_longer(cols = c("chromVAR_zscore_mean", "reporter_activity_minP"), names_to = "assay", values_to = "activity") %>% -->
<!--   mutate(activity_max = ave(activity, assay, FUN = max)) %>% -->
<!--   mutate(activity_norm = activity / activity_max) -->

<!-- ggplot(tf_activity_chromVAR_dev_norm %>%  -->
<!--          filter(tf %in% c("GATA4")), -->
<!--        aes(x = condition, y = activity)) + -->
<!--   geom_hline(yintercept = 0, linetype = "dashed") + -->
<!--   geom_bar(stat = "identity", width = .4) + -->
<!--   theme_pubr() + -->
<!--   facet_grid(assay~tf) -->
<!-- ``` -->


# Session Info
```{r}
paste("Run time: ",format(Sys.time()-StartTime))
getwd()
date()
sessionInfo()
```

